core/performance_test/benchmark/Cabana_NeighborArborXPerformance.cpp

```diff
@@ -0,0 +1,196 @@
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "Cabana_BenchmarkTimer.hpp"
+
+#include <Cabana_Core.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <vector>
+
+//---------------------------------------------------------------------------//
+// Performance test.
+template <class Device>
+void performanceTest( std::ostream& stream, const std::string& test_prefix )
+{
+    using exec_space = typename Device::execution_space;
+    using memory_space = typename Device::memory_space;
+
+    // Declare the neighbor list type.
+    using ListTag = Cabana::FullNeighborTag;
+    using IterTag = Cabana::SerialOpTag;
+
+    // Declare problem sizes.
+    double min_dist = 1.0;
+    std::vector<int> problem_sizes = { 1000, 10000, 100000, 1000000 };
+    int num_problem_size = problem_sizes.size();
+    std::vector<double> x_min( num_problem_size );
+    std::vector<double> x_max( num_problem_size );
+
+    // Declare the number of cutoff ratios (directly related to neighbors per
+    // atom) to generate.
+    std::vector<double> cutoff_ratios = { 4.0, 6.0 };
+    int cutoff_ratios_size = cutoff_ratios.size();
+
+    // Number of runs in the test loops.
+    int num_run = 10;
+
+    // Define the aosoa.
+    using member_types = Cabana::MemberTypes<double[3]>;
+    using aosoa_type = Cabana::AoSoA<member_types, Device>;
+    std::vector<aosoa_type> aosoas( num_problem_size );
+
+    // Create aosoas.
+    for ( int p = 0; p < num_problem_size; ++p )
+    {
+        int num_p = problem_sizes[p];
+        // Define problem grid.
+        x_min[p] = 0.0;
+        x_max[p] = 1.3 * min_dist * std::pow( num_p, 1.0 / 3.0 );
+        aosoas[p].resize( num_p );
+        auto x = Cabana::slice<0>( aosoas[p], "position" );
+        Cabana::Benchmark::createParticles( x, x_min[p], x_max[p], min_dist );
+
+        // Sort the particles to make them more realistic, e.g. in an MD
+        // simulation. They likely won't be randomly scattered about, but rather
+        // will be periodically sorted for spatial locality. Bin them in cells
+        // the size of the smallest cutoff distance.
+        double cutoff = cutoff_ratios.front() * min_dist;
+        double sort_delta[3] = { cutoff, cutoff, cutoff };
+        double grid_min[3] = { x_min[p], x_min[p], x_min[p] };
+        double grid_max[3] = { x_max[p], x_max[p], x_max[p] };
+        Cabana::LinkedCellList<Device> linked_cell_list( x, sort_delta,
+                                                         grid_min, grid_max );
+        Cabana::permute( linked_cell_list, aosoas[p] );
```

Comment:
Would be interesting to have an unsorted variant of this - I suspect ArborX would win in that case.

Reply:
I like that idea - easy to add a sorting bool